src/getDataDump.ts

```diff
@@ -75,81 +81,96 @@ export default async function getDataDump(
         }
     }
 
-    // to avoid having to load an entire DB's worth of data at once, we select from each table individually
-    // note that we use async/await within this loop to only process one table at a time (to reduce memory footprint)
-    while (tables.length > 0) {
-        const table = tables.shift()!
-
-        if (table.isView && !options.includeViewData) {
-            // don't dump data for views
-            retTables.push(merge<Table>([table, {
-                data: null,
-            }]))
+    try {
 
-            // eslint-disable-next-line no-continue
-            continue
+        if(options.lockTables) {
+            // see: https://dev.mysql.com/doc/refman/5.7/en/replication-solutions-backups-read-only.html
+            await executeSql(connection, `FLUSH TABLES WITH READ LOCK`)
+            await executeSql(connection, `SET GLOBAL read_only = ON`)
         }
 
-        currentTableLines = options.returnFromFunction ? [] : null
+        // to avoid having to load an entire DB's worth of data at once, we select from each table individually
```

Comment:
this whole block has just been indented by one level due to the added `try-catch` block